src/tools/SimpleTraversability.hpp

```diff
@@ -0,0 +1,192 @@
+        * obstacle if the maximum step around that cell is higher than the
+        * ground clearance
+        */
+        double groundClearance;
+
+        /**
+        * Traversable passages that are narrower (in m) than this will be
+        * closed
+        */
+        double minPassageWidth;
+
+        /**
+        * Clearance (in m) around obstacles. When set to a value larger than
+        * 0, all obstacles will be grown by a border of given width.
+        */
+        double obstacleClearance;
+
+        SimpleTraversabilityConfig()
+            : maximumSlope(0)
+            , classCount(0)
+            , groundClearance(0)
+            , minPassageWidth(0)
+            , obstacleClearance(0)
+        {
+        }
+
+        SimpleTraversabilityConfig(
+                double maximumSlope,
+                int classCount,
+                double groundClearance,
+                double minPassageWidth = 0,
+                double obstacleClearance = 0)
+
+            : maximumSlope(maximumSlope)
+            , classCount(classCount)
+            , groundClearance(groundClearance)
+            , minPassageWidth(minPassageWidth)
+            , obstacleClearance(obstacleClearance)
+        {
+        }
+
+    };
+
+    /**
+     * @brief Classification of terrain into symbolic traversability classes, based on
+     * geometric constraints
+     *
+     * For now, the modalities that are used are:
+     *
+     * <ul>
+     * <li>maximum slope
+     * <li>maximum step size
+     * </ul>
+     *
+     * The (very simple) model maps the local slope from [@a maximumSlope, 1] to [0, 1].
+     * The resulting value is then quantified in @a classCount intervals.
+     * Probability will be set to 1 for every classification beside CLASS_UNKNOWN.
+     *
+     * It outputs a TraversabilityGrid in which each cell has an integer value,
+     * this integer value being the traversability class for the cell. Since
+     * CLASS_UNKNOWN (=0) and CLASS_OBSTACLE (=1) are reserved, the resulting
+     * traversability class is in [2, 2 + classCount - 1]
+     *
+     * If one of the modalities is missing, it is simply ignored
+     */
+    class SimpleTraversability {
+
+        enum INPUT_DATA {
+            SLOPE,
+            MAX_STEP,
+            INPUT_COUNT
+        };
+
+    public:
+        enum CLASSES {
+            CLASS_UNKNOWN = 0,
+            CLASS_OBSTACLE = 1,
+            CUSTOM_CLASSES = 2
+        };
+    private:
+
+        SimpleTraversabilityConfig config;
+
+    public:
+        SimpleTraversability();
+
+        SimpleTraversability(
+                const SimpleTraversabilityConfig& config);
+
+        SimpleTraversability(
+                double maximumSlope,
+                unsigned int classCount,
+                double groundClearance,
+                double minPassageWidth = 0,
+                double obstacleClearance = 0);
+
+        void setConfig(const SimpleTraversabilityConfig& config);
```

Comment:
Not mandatory to change it now, but if there already is a config struct (like SimpleTraversabilityConfig) that should be sufficient to initialize or change the config.

Reply:
I added the function since the config is private (not sure if that is necessary, I left it like it was in the original) one was unable to change the config if for some reason the default constructor was used.